File: static/js/system_management.js
const API_BASE_URL = "";

let healthUpdateInterval = null;

/**
 * Start system health monitoring
 */
function startHealthMonitoring() {
    // Initial load
    updateSystemHealth();

    // Update every 5 seconds
    if (healthUpdateInterval) {
        clearInterval(healthUpdateInterval);
    }

    healthUpdateInterval = setInterval(() => {
        updateSystemHealth();
    }, 5000);
}

/**
 * Stop health monitoring
 */
function stopHealthMonitoring() {
    if (healthUpdateInterval) {
        clearInterval(healthUpdateInterval);
        healthUpdateInterval = null;
    }
}

/**
 * Update system health metrics
 */
async function updateSystemHealth() {
    try {
        const response = await fetch(`${API_BASE_URL}/api/admin/system/health`);
        const data = await response.json();

        if (data.success && data.metrics) {
            updateCPUMetrics(data.metrics.cpu);
            updateMemoryMetrics(data.metrics.memory);
            updateTemperatureMetrics(data.metrics.temperature);
            updateDiskMetrics(data.metrics.disk);
        }
    } catch (error) {
        console.error('Error updating system health:', error);
    }

    // Also update battery status
    await updateBatteryStatus();
}

/**
 * Update CPU metrics
 */
function updateCPUMetrics(cpu) {
    if (!cpu.available) return;

    const usage = cpu.usage;
    document.getElementById('cpuUsage').textContent = `${usage}%`;
    document.getElementById('cpuCores').textContent = `${cpu.cores} cores`;

    const bar = document.getElementById('cpuBar');
    bar.style.width = `${usage}%`;

    // Color based on usage
    if (usage < 50) {
        bar.style.background = '#10b981'; // Green
    } else if (usage < 80) {
        bar.style.background = '#f59e0b'; // Orange
    } else {
        bar.style.background = '#ef4444'; // Red
    }
}

/**
 * Update memory metrics
 */
function updateMemoryMetrics(memory) {
    if (!memory.available) return;

    const percent = memory.percent;
    document.getElementById('ramUsage').textContent = `${percent}%`;
    document.getElementById('ramInfo').textContent = `${memory.used_gb} GB / ${memory.total_gb} GB`;

    const bar = document.getElementById('ramBar');
    bar.style.width = `${percent}%`;

    // Color based on usage
    if (percent < 70) {
        bar.style.background = '#10b981'; // Green
    } else if (percent < 85) {
        bar.style.background = '#f59e0b'; // Orange
    } else {
        bar.style.background = '#ef4444'; // Red
    }
}

/**
 * Update temperature metrics
 */
function updateTemperatureMetrics(temp) {
    if (!temp.available) return;

    const temperature = temp.temp;
    document.getElementById('cpuTemp').textContent = `${temperature}°C`;

    const bar = document.getElementById('tempBar');
    const maxTemp = temp.critical || 100;
    const percent = Math.min((temperature / maxTemp) * 100, 100);
    bar.style.width = `${percent}%`;

    // Status and color
    let status, color;
    if (temperature < 60) {
        status = 'Normal';
        color = '#10b981'; // Green
    } else if (temperature < 70) {
        status = 'Warm';
        color = '#f59e0b'; // Orange
    } else if (temperature < 80) {
        status = 'Hot';
        color = '#ef4444'; // Red
    } else {
        status = 'Critical!';
        color = '#dc2626'; // Dark Red
    }

    document.getElementById('tempInfo').textContent = status;
    bar.style.background = color;
}

/**
 * Update disk metrics
 */
function updateDiskMetrics(disk) {
    if (!disk.available) return;

    const percent = disk.percent;
    document.getElementById('diskUsage').textContent = `${percent}%`;
    document.getElementById('diskInfo').textContent = `${disk.used_gb} GB / ${disk.total_gb} GB`;

    const bar = document.getElementById('diskBar');
    bar.style.width = `${percent}%`;

    // Color based on usage
    if (percent < 70) {
        bar.style.background = '#8b5cf6'; // Purple
    } else if (percent < 85) {
        bar.style.background = '#f59e0b'; // Orange
    } else {
        bar.style.background = '#ef4444'; // Red
    }
}

// Start health monitoring when page loads
document.addEventListener('DOMContentLoaded', function () {
    setTimeout(() => {
        startHealthMonitoring();
    }, 2000);
});

// Stop monitoring when leaving page
window.addEventListener('beforeunload', function () {
    stopHealthMonitoring();
});

// ==================== PHASE 5: DATA MANAGEMENT ====================

let currentCleanupAction = null;

/**
 * Load database statistics
 */
async function loadDatabaseStats() {
    try {
        const response = await fetch(`${API_BASE_URL}/api/admin/database/stats`);
        const data = await response.json();

        if (data.success && data.stats) {
            updateDatabaseStats(data.stats);
            displayCleanupSuggestions(data.suggestions || []);
        }
    } catch (error) {
        console.error('Error loading database stats:', error);
    }
}

/**
 * Update database statistics display
 */
function updateDatabaseStats(stats) {
    document.getElementById('dbTotalLogs').textContent = stats.total_logs || 0;
    document.getElementById('dbOldLogs').textContent = stats.old_logs_30d || 0;
    document.getElementById('dbDaysOfData').textContent = stats.days_of_data || 0;
}

/**
 * Display cleanup suggestions
 */
function displayCleanupSuggestions(suggestions) {
    const container = document.getElementById('cleanupSuggestions');

    if (!suggestions || suggestions.length === 0) {
        container.innerHTML = '';
        return;
    }

    const html = suggestions.map(s => `
        <div class="suggestion-card ${s.type}">
            <div class="suggestion-icon">${s.type === 'danger' ? '⚠️' : 'ℹ️'}</div>
            <div class="suggestion-content">
                <div class="suggestion-message">${s.message}</div>
                <button class="btn-sm" onclick="confirmCleanup('${s.action}', ${s.days})">Clean Up Now</button>
            </div>
        </div>
    `).join('');

    container.innerHTML = html;
}

/**
 * Show cleanup confirmation modal
 */
function confirmCleanup(action, days) {
    currentCleanupAction = { action, days: days || parseInt(document.getElementById('daysToKeep').value) };

    const messages = {
        'clear_all': 'Are you sure you want to delete ALL posture logs? This will permanently remove all historical data.',
        'clear_old': `Are you sure you want to delete logs older than ${currentCleanupAction.days} days?`,
        'clear_stats': 'Are you sure you want to delete all daily statistics? Charts will be empty until new data is collected.'
    };

    document.getElementById('cleanupMessage').textContent = messages[action] || 'Are you sure?';
    document.getElementById('confirmCleanup').checked = false;
    document.getElementById('confirmCleanupBtn').disabled = true;
    document.getElementById('cleanupModal').style.display = 'flex';
}

/**
 * Close cleanup modal
 */
function closeCleanupModal() {
    document.getElementById('cleanupModal').style.display = 'none';
    currentCleanupAction = null;
}

/**
 * Enable/disable confirm button based on checkbox
 */
document.addEventListener('DOMContentLoaded', function () {
    const checkbox = document.getElementById('confirmCleanup');
    const confirmBtn = document.getElementById('confirmCleanupBtn');

    if (checkbox && confirmBtn) {
        checkbox.addEventListener('change', function () {
            confirmBtn.disabled = !this.checked;
        });
    }

    // Load database stats on page load
    setTimeout(() => {
        loadDatabaseStats();
    }, 2000);
});

/**
 * Execute cleanup action
 */
async function executeCleanup() {
    if (!currentCleanupAction) return;

    const confirmBtn = document.getElementById('confirmCleanupBtn');
    confirmBtn.disabled = true;
    confirmBtn.textContent = 'Deleting...';

    try {
        const response = await fetch(`${API_BASE_URL}/api/admin/reset`, {
            method: 'POST',
            headers: { 'Content-Type': 'application/json' },
            body: JSON.stringify({
                confirm: true,
                action: currentCleanupAction.action,
                days: currentCleanupAction.days
            })
        });

        const data = await response.json();

        if (data.success) {
            alert(`Success! Deleted ${data.deleted} records.`);
            closeCleanupModal();
            loadDatabaseStats();
            loadData(currentPeriod); // Reload analytics data
        } else {
            alert(`Error: ${data.message}`);
        }
    } catch (error) {
        console.error('Error executing cleanup:', error);
        alert('Failed to execute cleanup');
    } finally {
        confirmBtn.disabled = false;
        confirmBtn.textContent = 'Confirm Delete';
    }
}

// ==================== BATTERY MONITORING ====================

console.log('Adding battery monitoring...');

/**
 * Update battery metrics
 */
async function updateBatteryStatus() {
    try {
        const response = await fetch(`${API_BASE_URL}/api/battery/latest`);
        const result = await response.json();

        if (result.success && result.data) {
            const battery = result.data;

            // Update display
            document.getElementById('batteryPercentage').textContent = `${Math.round(battery.percentage)}%`;
            document.getElementById('batteryVoltage').textContent = `${battery.voltage}V`;

            // Update level badge
            const levelBadge = document.getElementById('batteryLevelBadge');
            if (levelBadge) {
                levelBadge.textContent = battery.level.charAt(0).toUpperCase() + battery.level.slice(1);
                levelBadge.className = 'badge badge-' + battery.level;
            }

            // Update progress bar
            const bar = document.getElementById('batteryLevel');
            if (bar) {
                const pct = parseFloat(battery.percentage);
                bar.style.width = `${pct}%`;

                // Color based on percentage
                if (pct > 75) {
                    bar.style.background = '#10b981'; // Green
                } else if (pct > 50) {
                    bar.style.background = '#3b82f6'; // Blue
                } else if (pct > 25) {
                    bar.style.background = '#f59e0b'; // Orange
                } else {
                    bar.style.background = '#ef4444'; // Red
                }
            }

            // Update last update time
            const updateTime = new Date(battery.timestamp).toLocaleTimeString('vi-VN');
            const lastUpdateEl = document.getElementById('batteryLastUpdate');
            if (lastUpdateEl) {
                lastUpdateEl.textContent = updateTime;
            }
        }
    } catch (error) {
        console.error('Error updating battery status:', error);
    }
}

console.log('✓ Battery monitoring functions defined');

File: routes/system_management.py
"""
System Management Routes
Handles system health monitoring and data management
"""
from flask import Blueprint, render_template, request, jsonify
from flask_login import login_required
from utils.decorators import admin_required
from database import DBManager
from services.system_monitor import system_monitor
import logging

logger = logging.getLogger(__name__)

# Initialize database manager
db_manager = DBManager()

system_mgmt_bp = Blueprint('system_mgmt', __name__)


@system_mgmt_bp.route('/admin/system')
@login_required
@admin_required
def system_management():
    """System management page (health monitoring + data management)"""
    return render_template('admin_system.html')


# System Health API endpoints
@system_mgmt_bp.route('/api/admin/system/health')
@login_required
@admin_required
def get_system_health():
    """Get system health metrics"""
    try:
        metrics = system_monitor.get_metrics()
        return jsonify({'success': True, 'metrics': metrics}), 200
    except Exception as e:
        logger.error(f"Error getting system health: {e}", exc_info=True)
        return jsonify({'success': False, 'message': str(e)}), 500


# Database Stats API
@system_mgmt_bp.route('/api/admin/database/stats')
@login_required
@admin_required
def get_database_stats():
    """Get database statistics for cleanup suggestions"""
    try:
        stats = db_manager.get_database_stats()
        
        # Generate cleanup suggestions
        suggestions = []
        if stats.get('old_logs_30d', 0) > 1000:
            suggestions.append({
                'type': 'warning',
                'message': f"You have {stats['old_logs_30d']} logs older than 30 days. Consider cleaning them up.",
                'action': 'clear_old',
                'days': 30
            })
        
        if stats.get('total_logs', 0) > 50000:
            suggestions.append({
                'type': 'danger',
                'message': f"Database has {stats['total_logs']} logs. Performance may be affected.",
                'action': 'clear_old',
                'days': 60
            })
        
        return jsonify({
            'success': True,
            'stats': stats,
            'suggestions': suggestions
        }), 200
        
    except Exception as e:
        logger.error(f"Error getting database stats: {e}", exc_info=True)
        return jsonify({'success': False, 'message': 'Failed to get database stats'}), 500


# Data Reset API
@system_mgmt_bp.route('/api/admin/reset', methods=['POST'])
@login_required
@admin_required
def reset_data():
    """Clear historical data (with confirmation)"""
    try:
        data = request.get_json()
        confirm = data.get('confirm', False)
        action = data.get('action', 'clear_all')
        days = data.get('days', 30)
        
        if not confirm:
            return jsonify({'success': False, 'message': 'Confirmation required'}), 400
        
        logger.warning(f"Admin requested data reset: {action}")
        
        result = None
        if action == 'clear_all':
            result = db_manager.clear_all_logs()
        elif action == 'clear_old':
            result = db_manager.clear_old_logs(days=days)
        elif action == 'clear_stats':
            result = db_manager.clear_statistics()
        else:
            return jsonify({'success': False, 'message': 'Invalid action'}), 400
        
        if result and result.get('success'):
            return jsonify({
                'success': True,
                'message': f"Successfully deleted {result.get('deleted', 0)} records",
                'deleted': result.get('deleted', 0)
            }), 200
        else:
            return jsonify({
                'success': False,
                'message': result.get('error', 'Failed to delete data')
            }), 500
        
    except Exception as e:
        logger.error(f"Error resetting data: {e}", exc_info=True)
        return jsonify({'success': False, 'message': 'Failed to reset data'}), 500
